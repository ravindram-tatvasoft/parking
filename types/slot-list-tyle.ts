export interface SlotListType {
    id: string,
    isEmpty: boolean
    time: number | null
    exitTime: string
    slotNumber: number
    carNumber?: string
}