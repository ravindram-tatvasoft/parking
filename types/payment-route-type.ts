import { SlotListType } from "./slot-list-tyle";

export interface PaymentRouteType {
    slotDetail: SlotListType
}