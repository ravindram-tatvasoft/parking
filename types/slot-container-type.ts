import { SlotListType } from "./slot-list-tyle";

export interface SlotContainerType {
    item: SlotListType,
    index: number
    onRemoveCar: (index:number, item: SlotListType) => void
}