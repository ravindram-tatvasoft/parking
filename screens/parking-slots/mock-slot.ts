import moment from "moment";
import { SlotListType } from "../../types/slot-list-tyle";

export const mockDataSlots: SlotListType[] = [
    {
        id: 'id-1',
        exitTime: moment().format(),
        isEmpty: false,
        slotNumber: 1,
        time: 1,
        carNumber: "GJ 01 XX XXXX"
    },
    {
        id: 'id-2',
        exitTime: moment().format(),
        isEmpty: true,
        slotNumber: 2,
        time: 2,
        carNumber: "GJ 02 XX XXXX"
    }
]
