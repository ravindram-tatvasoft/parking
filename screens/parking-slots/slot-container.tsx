import { Dimensions, Image, StyleSheet, Text, TouchableOpacity, View } from "react-native"
import { images } from "../../assets/images/images"
import { SlotContainerType } from "../../types/slot-container-type"
import { SlotListType } from "../../types/slot-list-tyle"
import { colors } from "../../utils"

const width = Dimensions.get('screen').width
const viewSize = (width / 2) - 20


export const SlotContainer = (props: SlotContainerType) => {
    const { isEmpty, slotNumber } = props.item
    const parkingStatus = (item: SlotListType): string => {
        return `Slot ${slotNumber}\n ${item.isEmpty ? "Available" : "Not Available"}`
    }

    return (<TouchableOpacity disabled={isEmpty}
        style={styles.root}
        testID={`post-row-${props.index}`}
        onLongPress={() => props.onRemoveCar(props.index, props.item)}>
        <Text style={[styles.txtSlot, { color: isEmpty ? colors.green : colors.red }]}>{parkingStatus(props.item)}</Text>
        {!isEmpty && <View style={styles.carImgAndNumber}>
            <Text>{props.item.carNumber}</Text>
            <Image source={images.car} style={styles.carIcon} />
        </View>}
    </TouchableOpacity>)
}

const styles = StyleSheet.create({
    carImgAndNumber: {
        alignItems: "center"
    },
    root: {
        width: viewSize,
        height: viewSize,
        backgroundColor: colors.greyLight,
        margin: 10,
        alignItems: "center",
        justifyContent: 'center'
    },
    carIcon: {
        width: 50,
        height: 50,
        marginVertical: 10,
    },
    txtSlot: {
        textAlign: "center"
    }
})