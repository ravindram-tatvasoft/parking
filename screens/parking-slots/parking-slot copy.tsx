import { useEffect, useState } from 'react'
import { Alert, FlatList, StyleSheet, Text, TouchableOpacity, View } from 'react-native';
import { commonStyles, string } from '../../utils';
import { SlotContainer } from './slot-container';
import 'react-native-get-random-values';
import { v4 as uuidv4 } from 'uuid';
import { SlotListType } from '../../types/slot-list-tyle';
import { useDispatch, useSelector } from 'react-redux';
import { styles } from '../createlot/create-lot.style';
import { setParkingSlotList } from '../../redux/reducer/parking-slot-slice';
import { useNavigation } from '@react-navigation/core';
import moment from 'moment'


export default function ParkingSlot({ navigation }) {

  const parkingSlot = useSelector(state => state.parkingSlot);
  const dispatch = useDispatch()

  const removeCarAlert = (index: number, item: SlotListType) => {
    Alert.alert('', 'Are you sure you want to remove car from parking?', [
      { text: 'Cancel', onPress: () => null },
      { text: 'OK', onPress: () => removeCar(index, item) },
    ], { cancelable: false })
  }

  const removeCar = (index: number, item: SlotListType) => {
    let updateCar: SlotListType = { ...parkingSlot.slotList[index] };
    updateCar.exitTime = moment().format()
    const tempList = JSON.parse(JSON.stringify(parkingSlot.slotList));
    tempList.splice(index, 1, updateCar)
    dispatch(setParkingSlotList(tempList))
    navigation.navigate("PaymentScreen", { slotDetail: updateCar })
  }

  const renderItem = ({ item, index }) => (
    <SlotContainer item={item} index={index} onRemoveCar={removeCarAlert} />
  );

  return (
    <View style={commonStyles.container}>
      {parkingSlot.slotList.length > 0 ? <View>
        <Text>{string.noteRemoveCar}</Text>
        <FlatList
          data={parkingSlot.slotList}
          renderItem={renderItem}
          numColumns={2}
          keyExtractor={item => item.id}
        />
      </View> : <View style={commonStyles.viewCenter}>
        <Text>{string.emptySlot}</Text>
      </View>}
    </View>
  );
}

