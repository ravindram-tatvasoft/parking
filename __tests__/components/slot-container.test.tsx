// __tests__
import * as React from "react";
import { cleanup, render } from "@testing-library/react-native";
import { mockDataSlots } from "../../screens/parking-slots/mock-slot";
import { SlotContainer } from "../../screens/parking-slots/slot-container";
import { SlotContainerType } from "../../types/slot-container-type";

describe("Component: SlotContainer", () => {
    afterEach(() => {
        jest.clearAllMocks();
        cleanup()
    })

    test("should render ", () => {
        const props: SlotContainerType = {
            index: 1,
            item: mockDataSlots[0],
            onRemoveCar: jest.fn()
        }
        const tree = render(<SlotContainer {...props} />)
        expect(tree).toMatchSnapshot()
    })
})