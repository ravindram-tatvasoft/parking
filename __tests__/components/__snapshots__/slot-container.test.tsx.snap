// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: SlotContainer should render  1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#d5d5d5",
      "height": 355,
      "justifyContent": "center",
      "margin": 10,
      "opacity": 1,
      "width": 355,
    }
  }
  testID="post-row-1"
>
  <Text
    style={
      Array [
        Object {
          "textAlign": "center",
        },
        Object {
          "color": "red",
        },
      ]
    }
  >
    Slot 1
 Not Available
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
      }
    }
  >
    <Text>
      GJ 01 XX XXXX
    </Text>
    <Image
      source={1}
      style={
        Object {
          "height": 50,
          "marginVertical": 10,
          "width": 50,
        }
      }
    />
  </View>
</View>
`;
